transport/grpc/dial.go

```diff
@@ -186,7 +193,7 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 	// point when isDirectPathEnabled will default to true, we guard it by
 	// the Directpath env var for now once we can introspect user defined
 	// dialer (https://github.com/grpc/grpc-go/issues/2795).
-	if timeoutDialerOption != nil && o.EnableDirectPath && checkDirectPathEndPoint(endpoint) {
+	if timeoutDialerOption != nil && o.EnableDirectPath && checkDirectPathEndPoint(endpoint) && isOnComputeEngine() {
```

Comment:
Please use [metadata.OnGCE](https://pkg.go.dev/cloud.google.com/go/compute/metadata#OnGCE)

Reply:
Good to know!